Thanks for the credit! It's definitely a good consideration.

I've played around with the formatting to go for just Markdown in the way suggested. Is this the format you were aiming for?

# Test Report

*Report generated on 13-Mar-2019 at 23:49:44 by [pytest-md]*

[pytest-md]: https://github.com/hackebrot/pytest-md

## Summary

3 tests ran in 0.09 seconds

- 2 failed
- 1 passed


## Results

### tests/test_1.py

#### test_exception: Failed (0.00s)
```
def test_exception():
>       raise Exception("Failure\nmulti\nline")
E       Exception: Failure
E       multi
E       line

tests/test_1.py:6: Exception
```
#### test_pass: Passed (0.00s)

### tests/test_2.py

#### test_second_fail: Failed (0.00s)
```
def test_second_fail():
>       assert "foo" == "bar"
E       AssertionError: assert 'foo' == 'bar'
E         - foo
E         + bar

tests/test_2.py:2: AssertionError
```

--- a/src/pytest_md/plugin.py
+++ b/src/pytest_md/plugin.py
@@ -167,8 +167,8 @@ class MarkdownPlugin:
                         results += f" `{domaininfo}`\n"
 
                     if outcome in ("error", "failed"):
-                        results += f"```\n{report.longreprtext}\n```\n"
-                results += "\n"
+                        results += f"\n```\n{report.longreprtext}\n```\n"
+            results += "\n"
         return results
 
     def pytest_sessionfinish(self, session):
@@ -178,14 +178,18 @@ class MarkdownPlugin:
         header = self.create_header()
         project_link = self.create_project_link()
         summary = self.create_summary()
-        results = self.create_results()
 
         report = ""
         report += f"{header}\n\n"
         report += f"{project_link}\n\n"
-        report += f"{summary}\n\n"
-        report += f"{results}"
+        report += f"{summary}\n"
 
+        if self.config.option.verbose > 0:
+            results = self.create_results()
+            report += f"{results}"
+
+        # Cleanup trailing lines
+        report = f"{report.rstrip()}\n"
         self.report_path.write_text(report, encoding="utf-8")
 
 

--- a/src/pytest_md/plugin.py
+++ b/src/pytest_md/plugin.py
@@ -1,6 +1,5 @@
 import datetime
 import collections
-import html
 import pathlib
 import time
 
@@ -168,7 +167,7 @@ class MarkdownPlugin:
                         results += f" `{domaininfo}`\n"
 
                     if outcome in ("error", "failed"):
-                        results += f"```\n{report.longreprtext}```\n"
+                        results += f"```\n{report.longreprtext}\n```\n"
                 results += "\n"
         return results
 

--- a/src/pytest_md/plugin.py
+++ b/src/pytest_md/plugin.py
@@ -1,10 +1,11 @@
 import datetime
 import collections
+import html
 import pathlib
 import time
 
 
-OUTCOMES = ("failed", "passed", "skipped", "xfailed", "xpassed", "error")
+OUTCOMES = ("error", "failed", "passed", "skipped", "xfailed", "xpassed")
 
 
 class MarkdownPlugin:
@@ -131,6 +132,46 @@ class MarkdownPlugin:
 
         return summary + outcome_text
 
+    def create_results(self):
+        outcomes = collections.OrderedDict()
+
+        for outcome in OUTCOMES:
+            if outcome not in self.reports:
+                continue
+            file_reports = collections.OrderedDict()
+            for report in self.reports[outcome]:
+                filesystempath = report.location[0]
+                file_reports.setdefault(filesystempath, [])
+                file_reports[filesystempath].append(report)
+            outcomes[outcome] = file_reports
+
+        results = ""
+
+        for outcome, file_reports in outcomes.items():
+            outcome_text = outcome
+            if self.emojis_enabled:
+                outcome_text = self.emoji_repr[outcome].strip()
+
+            results += f"## {len(self.reports[outcome])} {outcome_text}\n\n"
+
+            for filesystempath, file_reports in file_reports.items():
+                results += f"### {filesystempath}\n\n"
+                for report in file_reports:
+                    domaininfo = report.location[2]
+                    results += f"{report.duration*100:.2f}s"
+                    if self.emojis_enabled:
+                        results += " ⏱ "
+
+                    if outcome == "error":
+                        results += f" `{outcome} at {report.when} of {domaininfo}`\n"
+                    else:
+                        results += f" `{domaininfo}`\n"
+
+                    if outcome in ("error", "failed"):
+                        results += f"```\n{report.longreprtext}```\n"
+                results += "\n"
+        return results
+
     def pytest_sessionfinish(self, session):
         self.session_finish = time.time()
         self.session_duration = self.session_finish - self.session_start
@@ -138,11 +179,13 @@ class MarkdownPlugin:
         header = self.create_header()
         project_link = self.create_project_link()
         summary = self.create_summary()
+        results = self.create_results()
 
         report = ""
         report += f"{header}\n\n"
         report += f"{project_link}\n\n"
-        report += f"{summary}"
+        report += f"{summary}\n\n"
+        report += f"{results}"
 
         self.report_path.write_text(report, encoding="utf-8")
 

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -151,12 +151,13 @@ def fixture_report_content(mode, now):
 
             8 tests ran in 0.00 seconds ⏱
 
+            - 1 💩
             - 1 😿
             - 3 🦊
             - 1 🙈
             - 1 🤓
             - 1 😜
-            - 1 💩"""
+            """
         )
 
     if mode is Mode.EMOJI_VERBOSE:
@@ -170,97 +171,75 @@ def fixture_report_content(mode, now):
 
             ## Summary
 
-            12 tests ran in 0.10 seconds ⏱
+            8 tests ran in 0.00 seconds ⏱
 
-                - 2 error 😡
-                - 1 failed 😰
-                - 3 passed 😃
-                - 2 skipped 🙄
-                - 2 xfail 😞
-                - 2 xpass 😲
+            - 1 error 💩
+            - 1 failed 😿
+            - 3 passed 🦊
+            - 1 skipped 🙈
+            - 1 xfailed 🤓
+            - 1 xpassed 😜
 
-            ## 2 error 😡
+            ## 1 ERROR 💩
 
-            ### test_example1.py
+            ### test_emoji_tests.py
 
-            0.02 s ⏱ `ERROR at setup of test_error`
+            0.00s ⏱  `error at setup of test_error`
 
             ```
-                @pytest.fixture
+            @pytest.fixture
                 def number():
             >       return 1234 / 0
             E       ZeroDivisionError: division by zero
 
-            test_example1.py:34: ZeroDivisionError
+            test_emoji_tests.py:37: ZeroDivisionError
             ```
 
-            ### test_example2.py
+            ## 1 FAILED 😿
 
-            0.02 s ⏱ `ERROR at setup of test_nope`
+            ### test_emoji_tests.py
 
-            ```
-                @pytest.fixture
-                def number():
-            >       return 1234 / 0
-            E       ZeroDivisionError: division by zero
-
-            test_example2.py:23: ZeroDivisionError
-            ```
-
-            ## 1 failed 😰
-
-            0.22 s ⏱ `test_failed`
+            0.00s ⏱  `test_failed`
 
             ```
-                def test_failed():
+            def test_failed():
             >       assert "emoji" == "hello world"
             E       AssertionError: assert 'emoji' == 'hello world'
             E         - emoji
             E         + hello world
 
-            test_example1.py:6: AssertionError
+            test_emoji_tests.py:5: AssertionError
             ```
 
-            ## 3 passed 😃
+            ## 3 PASSED 🦊
 
-            ### test_example1.py
+            ### test_emoji_tests.py
 
-            0.08 s ⏱ `test_passed[Sara-Hello Sara!]`
-            0.07 s ⏱ `test_passed[Mat-Hello Mat!]`
-            0.08 s ⏱ `test_passed[Annie-Hello Annie!]`
+            0.00s ⏱  `test_passed[Sara-Hello Sara!]`
+            0.00s ⏱  `test_passed[Mat-Hello Mat!]`
+            0.00s ⏱  `test_passed[Annie-Hello Annie!]`
 
-            ## 2 skipped 🙄
+            ## 1 SKIPPED 🙈
 
-            ### test_example1.py
+            ### test_emoji_tests.py
 
-            0.00 s ⏱ `test_skipped`
+            0.00s ⏱  `test_skipped`
 
-            ### test_example2.py
+            ## 1 XFAILED 🤓
 
-            0.00 s ⏱ `test_skipped`
+            ### test_emoji_tests.py
 
-            ## 2 xfail 😞
+            0.00s ⏱  `test_xfailed`
 
-            ### test_example1.py
+            ## 1 XPASSED 😜
 
-            0.40 s ⏱ `test_xfail`
+            ### test_emoji_tests.py
 
-            ### test_example2.py
-
-            0.20 s ⏱ `test_xfail`
-
-            ## xpass 😲
-
-            ### test_example1.py
-
-            0.42 s ⏱ `test_xpass`
-
-            ### test_example2.py
-
-            0.22 s ⏱ `test_xpass`
-"""
+            0.00s ⏱  `test_xpass`
+            """
         )
 
+    # Return the default report for Mode.NORMAL and Mode.VERBOSE
     if mode is Mode.VERBOSE:
         return textwrap.dedent(
             f"""\
@@ -274,15 +253,71 @@ def fixture_report_content(mode, now):
 
             8 tests ran in 0.00 seconds
 
+            - 1 error
             - 1 failed
             - 3 passed
             - 1 skipped
             - 1 xfailed
             - 1 xpassed
-            - 1 error"""
-        )
 
-    # Return the default report for Mode.NORMAL
+            ## 1 error
+
+            ### test_emoji_tests.py
+
+            0.00s `error at setup of test_error`
+
+            ```
+            @pytest.fixture
+                def number():
+            >       return 1234 / 0
+            E       ZeroDivisionError: division by zero
+
+            test_emoji_tests.py:37: ZeroDivisionError
+            ```
+
+            ## 1 failed
+
+            ### test_emoji_tests.py
+
+            0.00s `test_failed`
+
+            ```
+            def test_failed():
+            >       assert "emoji" == "hello world"
+            E       AssertionError: assert 'emoji' == 'hello world'
+            E         - emoji
+            E         + hello world
+
+            test_emoji_tests.py:5: AssertionError
+            ```
+
+            ## 3 passed
+
+            ### test_emoji_tests.py
+
+            0.00s `test_passed[Sara-Hello Sara!]`
+            0.00s `test_passed[Mat-Hello Mat!]`
+            0.00s `test_passed[Annie-Hello Annie!]`
+
+            ## 1 skipped
+
+            ### test_emoji_tests.py
+
+            0.00s `test_skipped`
+
+            ## 1 xfailed
+
+            ### test_emoji_tests.py
+
+            0.00s `test_xfailed`
+
+            ## 1 xpassed
+
+            ### test_emoji_tests.py
+
+            0.00s `test_xpass`
+            """
+        )
     return textwrap.dedent(
         f"""\
         # Test Report
@@ -295,12 +330,13 @@ def fixture_report_content(mode, now):
 
         8 tests ran in 0.00 seconds
 
+        - 1 error
         - 1 failed
         - 3 passed
         - 1 skipped
         - 1 xfailed
         - 1 xpassed
-        - 1 error"""
+        """
     )
 
 

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -156,8 +156,7 @@ def fixture_report_content(mode, now):
             - 1 🙈
             - 1 🤓
             - 1 😜
-            - 1 💩
-            """
+            - 1 💩"""
         )
 
     if mode is Mode.EMOJI_VERBOSE:
@@ -171,18 +170,119 @@ def fixture_report_content(mode, now):
 
             ## Summary
 
-            8 tests ran in 0.00 seconds ⏱
+            12 tests ran in 0.10 seconds ⏱
 
-            - 1 failed 😿
-            - 3 passed 🦊
-            - 1 skipped 🙈
-            - 1 xfailed 🤓
-            - 1 xpassed 😜
-            - 1 error 💩
-            """
+                - 2 error 😡
+                - 1 failed 😰
+                - 3 passed 😃
+                - 2 skipped 🙄
+                - 2 xfail 😞
+                - 2 xpass 😲
+
+            ## 2 error 😡
+
+            ### test_example1.py
+
+            0.02 s ⏱ `ERROR at setup of test_error`
+
+            ```
+                @pytest.fixture
+                def number():
+            >       return 1234 / 0
+            E       ZeroDivisionError: division by zero
+
+            test_example1.py:34: ZeroDivisionError
+            ```
+
+            ### test_example2.py
+
+            0.02 s ⏱ `ERROR at setup of test_nope`
+
+            ```
+                @pytest.fixture
+                def number():
+            >       return 1234 / 0
+            E       ZeroDivisionError: division by zero
+
+            test_example2.py:23: ZeroDivisionError
+            ```
+
+            ## 1 failed 😰
+
+            0.22 s ⏱ `test_failed`
+
+            ```
+                def test_failed():
+            >       assert "emoji" == "hello world"
+            E       AssertionError: assert 'emoji' == 'hello world'
+            E         - emoji
+            E         + hello world
+
+            test_example1.py:6: AssertionError
+            ```
+
+            ## 3 passed 😃
+
+            ### test_example1.py
+
+            0.08 s ⏱ `test_passed[Sara-Hello Sara!]`
+            0.07 s ⏱ `test_passed[Mat-Hello Mat!]`
+            0.08 s ⏱ `test_passed[Annie-Hello Annie!]`
+
+            ## 2 skipped 🙄
+
+            ### test_example1.py
+
+            0.00 s ⏱ `test_skipped`
+
+            ### test_example2.py
+
+            0.00 s ⏱ `test_skipped`
+
+            ## 2 xfail 😞
+
+            ### test_example1.py
+
+            0.40 s ⏱ `test_xfail`
+
+            ### test_example2.py
+
+            0.20 s ⏱ `test_xfail`
+
+            ## xpass 😲
+
+            ### test_example1.py
+
+            0.42 s ⏱ `test_xpass`
+
+            ### test_example2.py
+
+            0.22 s ⏱ `test_xpass`
+"""
         )
 
-    # Return the default report for Mode.NORMAL and Mode.VERBOSE
+    if mode is Mode.VERBOSE:
+        return textwrap.dedent(
+            f"""\
+            # Test Report
+
+            *Report generated on {report_date} at {report_time} by [pytest-md]*
+
+            [pytest-md]: https://github.com/hackebrot/pytest-md
+
+            ## Summary
+
+            8 tests ran in 0.00 seconds
+
+            - 1 failed
+            - 3 passed
+            - 1 skipped
+            - 1 xfailed
+            - 1 xpassed
+            - 1 error"""
+        )
+
+    # Return the default report for Mode.NORMAL
     return textwrap.dedent(
         f"""\
         # Test Report
@@ -200,8 +300,7 @@ def fixture_report_content(mode, now):
         - 1 skipped
         - 1 xfailed
         - 1 xpassed
-        - 1 error
-        """
+        - 1 error"""
     )
 
 

